python/lsst/daf/butler/core/datastore.py

```diff
@@ -76,6 +76,24 @@ def registerUndo(self, name, undoFunc, *args, **kwargs):
         """
         self._log.append(self.Event(name, undoFunc, args, kwargs))
 
+    @contextlib.contextmanager
+    def undoAs(self, name, undoFunc, *args, **kwargs):
```

Comment:
I was trying to indicate that the first argument (name) described the operation you were trying to undo, and that the second was the operation to undo it, but apparently that only made sense in my head.

I was also toying with `undoWith`; do either of those make more sense to you?  
```
with transaction.undoWith("do a thing", undoThing):
    doThing()
```
I think part of the problem is that the name of the operation is the first argument, but the real target of the context manager is `undoFunc`.  I could just swap them; unfortunately we can't really use kwargs on whatever argument isn't first because we need to be able to forward `*args`.

Any thoughts on how to make this better?

Reply:
I think I prefer `undoWith`.